server/data_common/matrix_loader.py

```diff
@@ -136,33 +136,56 @@ class MatrixDataType(Enum):
 
 
 class MatrixDataLoader(object):
-    def __init__(self, location, etype=None):
+    def __init__(self, location, matrix_data_type=None):
         """ location can be a string or DataLocator """
         self.location = DataLocator(location)
-        if etype is None:
-            self.etype = self.matrix_data_type()
-        else:
-            self.etype = etype
+        # matrix_data_type is an enum value of type MatrixDataType
+        self.matrix_data_type = matrix_data_type
+        # matrix_type is a DataAdaptor type, which corresonds to the matrix_data_type
         self.matrix_type = None
-        if self.etype == MatrixDataType.H5AD:
+
+        if matrix_data_type is None:
+            self.matrix_data_type = self.__matrix_data_type()
+
+        if self.matrix_data_type == MatrixDataType.H5AD:
             from server.data_anndata.anndata_adaptor import AnndataAdaptor
 
             self.matrix_type = AnndataAdaptor
-        elif self.etype == MatrixDataType.CXG:
+        elif self.matrix_data_type == MatrixDataType.CXG:
             from server.data_cxg.cxg_adaptor import CxgAdaptor
 
             self.matrix_type = CxgAdaptor
 
-    def matrix_data_type(self):
+    def __matrix_data_type(self):
         if self.location.path.endswith(".h5ad"):
             return MatrixDataType.H5AD
         elif ".cxg" in self.location.path:
             return MatrixDataType.CXG
         else:
             return MatrixDataType.UNKNOWN
 
+    def matrix_data_type_allowed(self, app_config):
```

Comment:
I'd rather see these checks done in the __init__ so it is impossible to get an instance if you have an unsupported/invalid/disabled data type.   Suggestion: have __init__ throw an "invalid data type" exception.